internal/pkg/aws/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -0,0 +1,178 @@
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+)
+
+const (
+	// SleepDuration is the sleep time for making the next request for log events.
+	SleepDuration        = 1 * time.Second
+	orderByLastEventTime = "LastEventTime"
+)
+
+var (
+	fatalCodes   = []string{"FATA", "FATAL", "fatal", "ERR", "ERROR", "error"}
+	warningCodes = []string{"WARN", "warn", "WARNING", "warning"}
+)
+
+type cloudwatchlogsClient interface {
+	DescribeLogStreams(input *cloudwatchlogs.DescribeLogStreamsInput) (*cloudwatchlogs.DescribeLogStreamsOutput, error)
+	GetLogEvents(input *cloudwatchlogs.GetLogEventsInput) (*cloudwatchlogs.GetLogEventsOutput, error)
+}
+
+// Service wraps an AWS Cloudwatch Logs client.
+type Service struct {
+	cwlogs cloudwatchlogsClient
+}
+
+// GetLogEventsOpts sets up optional parameters for LogEvents function.
+type GetLogEventsOpts func(*cloudwatchlogs.GetLogEventsInput)
+
+// WithLimit sets up limit for GetLogEventsInput
+func WithLimit(limit int) GetLogEventsOpts {
+	return func(in *cloudwatchlogs.GetLogEventsInput) {
+		in.Limit = aws.Int64(int64(limit))
+	}
+}
+
+// WithStartTime sets up startTime for GetLogEventsInput
+func WithStartTime(startTime int64) GetLogEventsOpts {
+	return func(in *cloudwatchlogs.GetLogEventsInput) {
+		in.StartTime = aws.Int64(startTime)
+	}
+}
+
+// WithEndTime sets up endTime for GetLogEventsInput
+func WithEndTime(endTime int64) GetLogEventsOpts {
+	return func(in *cloudwatchlogs.GetLogEventsInput) {
+		in.EndTime = aws.Int64(endTime)
+	}
+}
+
+// LogEventsOutput contains the output for LogEvents
+type LogEventsOutput struct {
+	// Retrieved log events.
+	Events []*Event
+	// Next tokens for each log stream.
+	NextTokens map[string]*string
+}
+
+// New returns a Service configured against the input session.
+func New(s *session.Session) *Service {
+	return &Service{
+		cwlogs: cloudwatchlogs.New(s),
+	}
+}
+
+// logStreams returns all name of the log streams in a log group.
+func (s *Service) logStreams(logGroupName string) ([]*string, error) {
+	resp, err := s.cwlogs.DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+		LogGroupName: aws.String(logGroupName),
+		Descending:   aws.Bool(true),
+		OrderBy:      aws.String(orderByLastEventTime),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("describe log streams of log group %s: %w", logGroupName, err)
+	}
+	if len(resp.LogStreams) == 0 {
+		return nil, fmt.Errorf("no log stream found in log group %s", logGroupName)
+	}
+	logStreamNames := make([]*string, len(resp.LogStreams))
+	for ind, logStream := range resp.LogStreams {
+		logStreamNames[ind] = logStream.LogStreamName
+	}
+	return logStreamNames, nil
+}
+
+// LogEvents returns an array of Cloudwatch Logs events.
+func (s *Service) LogEvents(logGroupName string, streamTokens map[string]*string, opts ...GetLogEventsOpts) (*LogEventsOutput, error) {
```

Comment:
Can we call this TaskLogEvents? I foresee a time in the future where we'll want to query logs that aren't particular to a task. For example, build logs, maybe?